src/umessage/umessagebuilder.rs

```diff
@@ -713,19 +669,15 @@ impl UMessageBuilder {
     /// # Ok(())
     /// # }
     /// ```
-    pub fn build_with_protobuf_payload<T: Message>(
+    pub fn build_with_protobuf_payload<T: MessageFull>(
         &mut self,
         payload: &T,
-    ) -> Result<UMessage, UMessageBuilderError> {
-        payload
-            .write_to_bytes()
-            .map_err(UMessageBuilderError::from)
-            .and_then(|serialized_payload| {
-                self.build_with_payload(
-                    serialized_payload.into(),
-                    UPayloadFormat::UPAYLOAD_FORMAT_PROTOBUF,
-                )
-            })
+    ) -> Result<UMessage, UMessageError> {
+        self.payload_format = UPayloadFormat::UPAYLOAD_FORMAT_PROTOBUF;
+        let mut message = self.build()?;
+        message
+            .insert_protobuf_payload(payload, Some(self.payload_format))
```

Comment:
it feels very strange that we use the builder to build the message and then _alter_ the message ...
Is there a particular reason why this builder's build functions cannot remain as they were? Why do we need `Message::insert_protobuf_payload`?